app/scripts/catalogue-sync.ts

```diff
@@ -0,0 +1,214 @@
+import { db } from "@/models";
+import { DataSourceCreationAttributes } from "@/models/DataSource";
+import env from "@next/env";
+import { randomUUID } from "node:crypto";
+
+const GLOBAL_API_URL =
+  process.env.NODE_ENV === "production"
+    ? "https://ccglobal.citycatalyst.io"
+    : "https://ccglobal.openearth.dev";
+
+interface Source {
+  datasource_id: string;
+  name: string;
+  source_type: string;
+  url: string;
+  description: string;
+  access_type: string;
+  geographical_location: string;
+  start_year: number;
+  end_year: number;
+  latest_accounting_year: number;
+  frequency_of_update: string;
+  spatial_resolution: string;
+  language: string;
+  accessibility: string;
+  data_quality: string;
+  notes: string;
+  units: string;
+  methodology_url: string;
+  publisher_id: string;
+  retrieval_method: string;
+  api_endpoint: string;
+  gpc_reference_number?: string;
+  created_date?: string;
+  modified_date?: string;
+  subcategory_id?: string;
+  subsector_id?: string;
+  created?: Date;
+  last_updated?: Date;
+}
+
+function snakeToCamel(str: string): string {
+  return str
+    .toLowerCase()
+    .replace(/([-_][a-z])/g, (group) =>
+      group.toUpperCase().replace("-", "").replace("_", ""),
+    );
+}
+
+async function syncDataCatalogue() {
+  const projectDir = process.cwd();
+  env.loadEnvConfig(projectDir);
+  if (!db.initialized) {
+    await db.initialize();
+  }
+
+  let catalogue = await db.models.Catalogue.findOne({
+    where: { type: "global_api" },
+  });
+  if (!catalogue) {
+    catalogue = await db.models.Catalogue.create({
+      type: "global_api",
+      lastUpdate: new Date(0), // UNIX epoch as default value
+    });
+  }
+  const previousUpdate = catalogue.lastUpdate?.getTime() || 0;
+
+  const lastUpdateResponse = await fetch(
+    `${GLOBAL_API_URL}/api/v0/catalogue/last-update`,
+  );
+  const lastUpdate = await lastUpdateResponse.json();
+  if (!lastUpdate?.last_update) {
+    throw new Error(
+      "Failed to query last catalogue update with error " +
+        lastUpdateResponse.status +
+        " " +
+        lastUpdateResponse.statusText,
+    );
+  }
+
+  console.log(
```

Comment:
Are we using this for logging?

Reply:
I think it's helpful to know the timestamps for debugging purposes in case the catalogue doesn't update properly on the server.